src/main/java/org/apache/mesos/hdfs/Scheduler.java

```diff
@@ -545,43 +545,39 @@ private boolean offerNotEnoughResources(Offer offer, double cpus, int mem) {
   }
 
   private void reconcileTasks(SchedulerDriver driver) {
-    updateReconciliationDeadline();
     driver.reconcileTasks(Collections.<Protos.TaskStatus> emptyList());
-    reconcilePersistentState();
-    reconciliationCompleted = true;
+    Timer timer = new Timer();
+    timer.schedule(new ReconcileStateTask(), conf.getReconciliationTimeout() * 1000);
```

Comment:
Understood, let me add a TODO to do periodic reconciles with exponential backoffs to take care of this scenario. Right now, there is one but it shouldn't be terribly complex to repeat the process in the case of master failover and waiting for slaves to reregister.